File: C++/Longest-Consecutive-Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> hash;
        for (int i = 0; i < nums.size(); i ++) {
            if (hash.find(nums[i]) == hash.end()) {
                hash.insert(nums[i]);
            }
        }
        int maxlen = 0;
        for (int i = 0; i < nums.size(); i ++) {
            if (hash.find(nums[i]) == hash.end()) continue;
            int cur = 1, cnt = nums[i] - 1;
            hash.erase(nums[i]);
            while (hash.find(cnt) != hash.end()) {
                cur ++;
                hash.erase(cnt--);
            }
            cnt = nums[i] + 1;
            while (hash.find(cnt) != hash.end()) {
                cur ++;
                hash.erase(cnt++);
            }
            maxlen = max(maxlen, cur);
        }
        return maxlen;
    }
};

File: Python/Longest-Consecutive-Sequence.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def longestConsecutive(self, num):
        if not num:
            return 0
        seq = {}
        for i in num:
            if i not in seq:
                seq[i] = 1
        res = 1
        for i in num:
            cnt = i - 1
            cur = 1
            while cnt in seq:
                del seq[cnt]
                cnt -= 1
                cur += 1
            cnt = i + 1
            while cnt in seq:
                del seq[cnt]
                cnt += 1
                cur += 1
            res = max(res, cur)
        return res